### PR TITLE
WIP: pocket as separate drawer

### PR DESCRIPTION
**Pull request by zippy**
Initial work to see what it feels like to have the pocket as a drawer.

### CODE REVIEW

File: src/renderer/src/elements/main-dashboard.ts [hunk 1]
```diff
@@ -1157,6 +1216,26 @@ export class MainDashboard extends LitElement {
             </button>
           </sl-tooltip>
         </div>
+        <div class="row center-content" style="margin-bottom: 5px;">
+          <sl-tooltip content="Search" placement="right" hoist>
```
Comment (matthme) [inline src/renderer/src/elements/main-dashboard.ts:1220]: content="Pocket"

File: src/renderer/src/elements/main-dashboard.ts [hunk 2]
```diff
@@ -1157,6 +1216,26 @@ export class MainDashboard extends LitElement {
...
+            <button
+              class="moss-button"
+              @click=${() => this.openPocket()}
```
Comment (matthme) [inline src/renderer/src/elements/main-dashboard.ts:1223]: How about changing this to `togglePocket()`? I'd find it more intuitive if the pocket closes again when clicking the button twice.

File: src/renderer/src/elements/pocket.ts [hunk 3]
```diff
@@ -140,14 +129,10 @@ export class MossPocket extends LitElement {
...
-        style="--width: 800px;"
-        no-header
-        @sl-initial-focus=${(e: { preventDefault: () => void }) => {
-          e.preventDefault();
-          this._searchField.focus();
-        }}
+      <sl-drawer
+        placement="bottom" class="drawer-contained"
+        id="pocket"
+        contained
```
Comment (matthme) [inline src/renderer/src/elements/pocket.ts:135]: The `contained` attribute here seems to disable close on side-click. Do you want to not have it close on side-click intentionally? If so I think I would personally prefer to have the pocket have less height in order to not cover as much tool view port.

File: src/renderer/src/elements/pocket.ts [hunk 4]
```diff
@@ -157,40 +142,28 @@ export class MossPocket extends LitElement {
...
+            <img src="pocket_black.png" style="height: 30px; margin-right: 10px;">
+                ${msg('Pocket:')}
+            <div style="display:inline-block;font-size:80%;color: var(--sl-color-secondary-950)">
+              <span style="background: #e0e0e0; padding: 2px 5px; border-radius: 4px; color: black;"
+                        >Alt + P</span>
+              ${msg('to open')}
+            </div>
+          </div>
+        </div>
+        <sl-icon-button slot="header-actions"
```
Comment (matthme) [inline src/renderer/src/elements/pocket.ts:155]: A tooltip would be nice here to understand the consequences of clicking it.

File: src/renderer/src/elements/main-dashboard.ts [hunk 5]
```diff
@@ -500,13 +508,25 @@ export class MainDashboard extends LitElement {
...
-            this.showClipboard = true;
+            this.showSearch = true;
+            this._searchDialog.show('open');
+            this._searchDialog.focus();
+            break;
+          case true:
+            this._searchDialog.hide();
+            break;
+        }
+      } else if (zEvent.altKey && zEvent.key === 'p') {
```
Comment (matthme) [inline src/renderer/src/elements/main-dashboard.ts:525]: I can't press `Alt + P` with a single hand on my keyboard...I wonder whether we could choose a keyboard combination that's easy to press with a single hand.

**Review by matthme — CHANGES_REQUESTED** [review]
Added a few comments. In general I think it's worth trying it out in a drawer and see how it feels.

File: we.dev.config.example.ts [hunk 6]
```diff
@@ -86,12 +86,12 @@ export default defineConfig({
       subtitle: 'Collaborative note taking',
       description: 'Real-time notetaking based on syn',
       icon: {
-        type: 'https',
-        url: 'https://lightningrodlabs.org/projects/notebooks.png',
+        type: 'filesystem',
```
Comment (matthme) [inline we.dev.config.example.ts:89]: Oh, and can you undo those changes to the example config? The example config is intended to work out of the box when cloning the repository so it should not contain references to local paths. For developing on Moss I use a dedicated `we.dev.config.ts` file which is used by the `yarn applet-dev` and `yarn applet-dev-1` commands and which is in the `.gitignore` file.

File: src/renderer/src/elements/pocket.ts (hunk 3 above)
Comment (zippy) [reply src/renderer/src/elements/pocket.ts:135]: hmmm.  I thought that you might want to be able to keep the pocket open while doing other things.  If you don't do contained that it's a modal.  But I'm not sure about that.

File: src/renderer/src/elements/main-dashboard.ts (hunk 5 above)
Comment (zippy) [reply src/renderer/src/elements/main-dashboard.ts:525]: Alt-e or Alt-t?

File: we.dev.config.example.ts (hunk 6 above)
Comment (zippy) [reply we.dev.config.example.ts:89]: whoops that was a mistake.  Ah, that's a good idea.

File: src/renderer/src/elements/pocket.ts (hunk 4 above)
Comment (zippy) [reply src/renderer/src/elements/pocket.ts:155]: Interesting, I couldn't get sl-tooltip to work.  I had to use `title=""`..

File: src/renderer/src/elements/pocket.ts (hunk 3 above)
Comment (matthme) [reply src/renderer/src/elements/pocket.ts:135]: Okay that makes sense. But I have the sense that we should have it cover less vertical space then, what do you think?

**Pull request closed without merging**